### PR TITLE
Update to v8.3.1

### LINKED ISSUE #56: Are the 8.3.x releases supported?
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

It seems the latest [ESMF/ESMPy release is 8.3.1 (released 21 Sept 2022)](http://earthsystemmodeling.org/esmpy_doc/release/ESMF_8_3_1/html/), but the latest version available on [conda-forge is 8.2.0 (released 28 Oct 2021)](https://anaconda.org/conda-forge/esmpy/files). Should we expect support for the more recent releases ASAP? Is something holding it up?

### Installed packages

```shell
Nil
```

### Environment info

```shell
Nil
```

### PR DESCRIPTION
**Pull request by xylar**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

closes #56

### CODE REVIEW

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
`nompi` tests are failing because they require PIO:
https://github.com/esmf-org/esmf/blob/79f77aa83033b2fbaab1754b0e1c2968acf4ac48/src/Infrastructure/Mesh/src/ESMCI_Mesh_FileIO.C#L296-L298

We'll presumably need to patch to disable some tests for nompi

OSX MPICH failing because that ESMF library failed to build -- trying again.  OSX OpenMPI, no idea yet...

File: recipe/meta.yaml [hunk 1]
```diff
@@ -15,8 +14,11 @@ package:
   version: {{ version }}
 
 source:
-  url: https://github.com/esmf-org/esmf/archive/ESMF_{{ ver }}/esmf_{{ ver }}_src.tar.gz
-  sha256: 3693987aba2c8ae8af67a0e222bea4099a48afe09b8d3d334106f9d7fc311485
+  url: https://github.com/esmf-org/esmf/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 6c39261e55dcdf9781cdfa344417b9606f7f961889d5ec626150f992f04f146d
+  patches:
+    - 0001-remove-tests-for-no-pio.patch  # [mpi == 'nompi']
+    - 0002-remove-failing-osx-test.patch  # [osx]
```
Comment (xylar) [inline recipe/meta.yaml:21]: I don't feel great about this but it doesn't seem worth putting too much time into unless it happens with the newly released 8.4.0 as well.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -15,8 +14,11 @@ package:
   version: {{ version }}
 
 source:
-  url: https://github.com/esmf-org/esmf/archive/ESMF_{{ ver }}/esmf_{{ ver }}_src.tar.gz
-  sha256: 3693987aba2c8ae8af67a0e222bea4099a48afe09b8d3d334106f9d7fc311485
+  url: https://github.com/esmf-org/esmf/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 6c39261e55dcdf9781cdfa344417b9606f7f961889d5ec626150f992f04f146d
+  patches:
+    - 0001-remove-tests-for-no-pio.patch  # [mpi == 'nompi']
+    - 0002-remove-failing-osx-tests.patch  # [osx]
```
Comment (xylar) [inline recipe/meta.yaml:21]: Definitely worth finding out why cyclic decomposition isn't working in OSX if this is still a problem in 8.4.0

**Comment by xylar** [comment]
Okay, too may OSX test are failing.  I think something is wrong with the ESMF OSX builds.

**Comment by xylar** [comment]
@ocefpaf and @rokuingh, I'm a bit stuck. There seems to be a library incompatibility here vs in the corresponding ESMF build but I don't see it.  This is holding up Python 3.11 migration so it would be great to figure out.

Anyone we can bring in that would be able to figure this out more quickly?

**Comment by ocefpaf** [comment]
> Anyone we can bring in that would be able to figure this out more quickly?

Looking into it...

**Comment by xylar** [comment]
@conda-forge-admin, please restart ci

**Comment by xylar** [comment]
@ocefpaf, I tried rebuilding ESMF in https://github.com/conda-forge/esmf-feedstock/pull/84 and then verified just now that this build of ESMPy is using all the same host libraries as that build.  But still no luck.  I'm kind of relieved because that would have been a strange issue to debug but I'm still stumped.

Any suggestions of who I can turn to for help?

**Comment by ocefpaf** [comment]
macOS is not my expertise. @isuruf I know you are super busy with your thesis but if you believe this is a low hanging fruit do you mind giving us some guidance here?

**Comment by xylar** [comment]
@zklaus, does this happen to be one you can help out with?  We can't migrate to OSX ARM here in #60 until we first update here.

**Comment by zklaus** [comment]
@conda-forge-admin, please rerender

**Comment by zklaus** [comment]
After working with #59, I am fairly certain that the segfaults come from bad tests/a certain incompatibility between tests and the c/c++/fortran library. We can deal with them probably best in the same way as in #59, but I'll need to inspect some artifacts to confirm.

**Comment by xylar** [comment]
@zklaus, a question is if it's even worth getting 8.3.1 to build or if we skip it and go to 8.4.0.

**Comment by zklaus** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
@zklaus, I'm fine with that.  Hopefully, not that many people will need this version if 8.4.0 is available at the same time.

**Comment by xylar** [comment]
Please feel free to merge this and #59 when you're happy.

**Comment by xylar** [comment]
And thank you very, very much for the hard work on these!

**Comment by zklaus** [comment]
Cheers, @xylar and thanks to you too!

**Pull request merged**